xbitinfo/xbitinfo.py

```diff
@@ -487,9 +487,13 @@ def get_cdf_without_artificial_information(
                     infbits = i
                     break
 
-            for i in range(infbits + 1, len(cdf_array)):
-                cdf_array[i] = 0
+            for i in range(0, infbits + 1):
+                # Normalize CDF values for elements up to 'infbits'.
+                cdf_array[i] = cdf_array[i] / cdf_array[infbits]
```

Comment:
We don't need a loop here and can just divide the entire array by its sum
```suggestion
            cdf_array = cdf_array / cdf_array[infbits]
```